documentation/grammars.md

```diff
@@ -6,57 +6,111 @@ Grammars in Pliant are defined by an bnf dialect similar to abnf and ebnf. Some
+
+lexer rule expression =  
+      lexer rule term
+    | lexer rule term, '|', lexer rule expression ;
+
+lexer rule term       =   
+      lexer rule factor
+    | lexer rule factor, lexer rule term ;
+
+lexer rule factor     =   
+      literal
+    | '/' regular expression '/' ;
+
+letter = 
+      "A" | "B" | "C" | "D" | "E" | "F" | "G"
+    | "H" | "I" | "J" | "K" | "L" | "M" | "N"
+    | "O" | "P" | "Q" | "R" | "S" | "T" | "U"
+    | "V" | "W" | "X" | "Y" | "Z" | "a" | "b"
+    | "c" | "d" | "e" | "f" | "g" | "h" | "i"
+    | "j" | "k" | "l" | "m" | "n" | "o" | "p"
+    | "q" | "r" | "s" | "t" | "u" | "v" | "w"
+    | "x" | "y" | "z" ;
+       
+digit = 
+      "0" | "1" | "2" | "3" | "4" 
+    | "5" | "6" | "7" | "8" | "9" ;
+
+letter or digit = 
+    letter | digit ;
+
+(* 
+    https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form#Extensibility 
+    Defining "? regex {regular expression} ?" where {regular expression} is a POSIX compliant regular expression.
+*)
+
+single quote string = 
+    ? regex ['][^']*['] ?;
+
+double quote string = 
+    ? regex ["][^"]*["] ?;
+
+(* whitespace *)
+ws = 
+    ? regex \s+ ?;                    
 ```
-Definition          =   Block
-                        | Block Definition
 
-Block               =   Rule
-                        | Setting
-                        | LexerRule
+# Predefined Settings
 
-Rule                =   QualifiedIdentifier '=' Expression ';'
-
-Setting             =   SettingIdentifier '=' QualifiedIdentifier ';'
-
-LexerRule           =   QualifiedIdentifier '~' LexerRuleExpression ';'
-
-Expression          =   Term
-                        | Term '|' Expression
-
-Term                =   Factor
-                        | Factor Term
-
-Factor              =   QualifiedIdentifier
-                        | Literal
-                        | '/' RegularExpression '/'
-                        | Repetition
-                        | Optional
-                        | Grouping
-
-Literal             =   SingleQuoteStringLexerRule
-                        | DoubleQuoteStringLexerRule
-
-Repetition          =   '{' Expression '}'
-
-Optional            =   '[' Expression ']'
-
-Grouping            =   '(' Expression ')'
-
-QualifiedIdentifier =   Identifier
-                        | Identifier '.' QualifiedIdentifier
-
-LexerRuleExpression =   LexerRuleTerm
-                        | LexerRuleTerm '|' LexerRuleExpression
-
-LexerRuleTerm       =   LexerRuleFactor
-                        | LexerRuleFactor LexerRuleTerm
-
-LexerRuleFactor     =   Literal
-                        | '/' RegularExpression '/'
-
-SingleQuoteStringLexerRule ~ /['][^']*[']/
-
-DoubleQuoteStringLexerRule ~ /["][^"]*["]/
-
-```
+Settings are used to add meta information to the parser. The following settings are recognized in pliant.
 
+| Setting | Description | Example Use |
+| ------- | ----------- | ----------- |
+| start   | denotes the start symbol | :start S ;|
```

Comment:
shouldn't Pliant's grammar itself have 3 start symbols (for bnf, ebnf and abnf) ?